crates/lsp/src/lsp.rs

```diff
@@ -696,6 +696,10 @@ impl LanguageServer {
                         did_save: Some(true),
                         ..TextDocumentSyncClientCapabilities::default()
                     }),
+                    diagnostic: Some(DiagnosticClientCapabilities {
+                        dynamic_registration: Some(true),
+                        related_document_support: Some(true),
```

Comment:
Alternatively, we could specify that Zed does not support handling diagnostics for related documents currently with `Some(false)`.

Reply:
>What's the difference from Zed's perspective from setting this to false or true?

The difference is, with `true`, Zed tells a server that it can process diagnostics for related documents when it receives diagnostics.

>For the related documents, we would get the diagnostics in the same format, but for different files and with different metadata?

Yes, correct.

>If that's it, we seem to be able to display these too, and can support this capability?

Pretty much, yes. I will implement that feature.